scripts/video_summary.py

```
@@ -0,0 +1,108 @@
+import pytesseract
+import openai
+
+
+def download_video(gdrive_url: str, output_dir: str = "downloads") -> str:
+    """Download a video from Google Drive link and return local path."""
+    os.makedirs(output_dir, exist_ok=True)
+    if 'drive.google.com' in gdrive_url:
+        try:
+            file_id = gdrive_url.split('/d/')[1].split('/')[0]
+        except IndexError:
+            raise ValueError('Invalid Google Drive URL')
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)